### DISCUSSION TITLE
Auto configuration of the java agent and adding custom pass through propagator

### DISCUSSION THREAD

**Discussion by grahamia** [open-telemetry/opentelemetry-java-instrumentation#9489] [Q&A] [answered]
I am trying to propagate custom headers in a application using the PassThroughPropagator (following the changes made here - https://github.com/typelevel/otel4s/issues/277).

I had it all working by instanciating the otel instance within code, i.e. 
```scala
AutoConfiguredOpenTelemetrySdk
        .builder()
        .addPropagatorCustomizer { (autoConfigured, _) =>
          TextMapPropagator.composite(autoConfigured,
                                      PassThroughPropagator.create(propagationConfig.ciHeaders.map(_.toString).asJava))
        }
...
```

But then when trying to enable java agent to get jvm metrics etc I obviously ran into issue of agent starting the global instance and then the code also starting it, even with setting global to false still run into issue of multiple prometheus servers starting (yes, want prometheus exporter).

So after a lot of digging came across `ConfigurablePropagatorProvider` but then had to struggle to get that working in a `otel.javaagent.extensions` as the jars provided here seem to have to be self container and I assume since `PassThroughPropagator` is in incubator it isn't within the javaagent jar I had to go down the path of creating a fat jar containing my simple `PassThroughConfigurablePropagator` and include the `opentelemetry-extension-incubator`. Then in code just do a simple `GlobalOpenTelemetry.get` to obtain a handle to the global otel instance.

Is there any plan on providing a `PassThroughConfigurablePropagator` like other propagators that can be created through config?

My simple code is as follows:
```java
class PassThroughConfigurablePropagator implements ConfigurablePropagatorProvider {

    @Override
    public TextMapPropagator getPropagator(ConfigProperties config) {
        return PassThroughPropagator.create(config.getList("otel.passthrough.headers"));
    }

    @Override
    public String getName() {
        return "passthrough";
    }
}
```

The above can then be used by adding `passthrough` to the env var list `OTEL_PROPAGATORS` and defining a list of headers in new env var `OTEL_PASSTHROUGH_HEADERS`

**Comment by trask** [comment, marked as answer]
hi @grahamia! we'd probably want to see https://github.com/open-telemetry/opentelemetry-specification/issues/1717 make it into the specification, and the name defined in the specification under `OTEL_PROPAGATORS` before adding support for it in the out-of-the-box Java agent

**Comment by grahamia** [comment]
Thanks @trask glad I wasn't incorrect in how I got round the issues, will use what I have implemented and hopefully in the future the implementation within OpenTelemetry isnt a million miles different.
